### PR TITLE
Fix warnings from grep

### PR DESCRIPTION
**Pull request by c---**
Newer versions of grep will issue a warning for invalid "escaped" slashes which in turn causes a bunch of warnings to appear while building.
```
grep (GNU grep) 3.8
grep: warning: stray \ before /
```

### CODE REVIEW

**Comment by c---** [comment]
This should probably be applied to openwrt-22.03 as well. Should I create another PR?

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ $(if $(findstring $(space),$(TOPDIR)),$(error ERROR: The path to the OpenWrt dir
 
 world:
 
-DISTRO_PKG_CONFIG:=$(shell $(TOPDIR)/scripts/command_all.sh pkg-config | grep -E '\/usr' -m 1)
```
Comment (Ansuel) [inline Makefile:17]: Should we drop the `-E` ?

File: target/sdk/files/Makefile [hunk 2]
```diff
@@ -14,7 +14,7 @@ export TOPDIR LC_ALL LANG SDK
 
 world:
 
-DISTRO_PKG_CONFIG:=$(shell $(TOPDIR)/scripts/command_all.sh pkg-config | grep -E '\/usr' -m 1)
```
Comment (Ansuel) [inline target/sdk/files/Makefile:17]: Should we drop the `-E` ?

**Comment by c---** [comment]
@Ansuel What is that about markdown?

Hopefully I have split the commits correctly. Previously I thought maybe `-E` was reserved for future use but I guess there is no reason to keep it if will not be used. How does this look?

**Comment by markmentovai** [comment]
See #10670.

File: target/sdk/files/Makefile (hunk 2 above)
Comment (httpstorm) [reply target/sdk/files/Makefile:17]: `-E` is for *extended regular expression* according to man, so we don't need it.

**Comment by httpstorm** [comment]
I performed some test builds on macOS on fresh git clone. Works fine.

File: target/sdk/files/Makefile (hunk 2 above)
Comment (Ansuel) [reply target/sdk/files/Makefile:17]: yep that was the idea... no idea why it was used on the first place?

**Comment by Ansuel** [comment]
@httpstorm can you provide tested-by tag?

@c---  this part is not supported by git and should be removed or simply remove the ```
```
grep (GNU grep) 3.8
grep: warning: stray \ before /
```

**Comment by c---** [comment]
@Ansuel  Sorry, I thought I had removed that. I had to push an amend twice, hopefully it works out this time.

**Review by neheb — APPROVED** [review]

**Comment by Ansuel** [comment]
@c--- can you rebase this and add the tested by tag?

Also the base-files commit should be dropped... The file is run under an openwrt system where busybox grep is used... Also this should be related to build warning not warning on running system.

**Comment by c---** [comment]
Merged with latest master and rebased the whole pull request. Added Tested-by.

Related to build warning? Is that a different indicator than `build:`?

**Pull request merged**